### PR TITLE
[develop][REVIEW] Trello-PDraWvOL: Added error handling via express middleware

### PR DESCRIPTION
**Pull request by AndreySurzhan**
[Trello-PDraWvOL](https://trello.com/c/PDraWvOL): Added error handling via express middleware

- [x] Rebase

### CODE REVIEW

**Comment by AndreySurzhan** [comment]
@wildeastengineer please review

**Comment by AndreySurzhan** [comment]
Right now we have the following error handling behavior:
send 404 not found error - incorrect route path or given rout entity was not found
send 400 bad request - validation error (ref ids were not found)
send 405 method not allowed - route exist but method for this rout is incorrect
send 500 unhandled server error - send in other unhandled cases

**Comment by AndreySurzhan** [comment]
I am going to impalement this in scope of that task 
https://trello.com/c/NYYge5vc

**Comment by AndreySurzhan** [comment]
@wildeastengineer could you please review
(https://trello.com/c/NYYge5vc [DB] Handle mongoose CastErorr and cast it as validation error)

File: utils/expressPlugins/notFoundErrorHandler.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+var _ = require('underscore');
```
Comment (wildeastengineer) [inline utils/expressPlugins/notFoundErrorHandler.js]: `_` is not used

**Comment by wildeastengineer** [comment]
except that :+1:

**Pull request merged**